src/esgvoc/api/data_descriptors/source.py

```diff
@@ -1,35 +1,73 @@
-from typing import Optional
-
-from pydantic import Field
+"""
+Model (i.e. schema/definition) of the source descriptor
+"""
 
 from esgvoc.api.data_descriptors.data_descriptor import PlainTermDataDescriptor
-from esgvoc.api.data_descriptors.activity import Activity
+from esgvoc.api.data_descriptors.model_component_new import EMDModelComponent
+from esgvoc.api.data_descriptors.organisation import Organisation
 
 
 class Source(PlainTermDataDescriptor):
     """
-    A 'source' refers to a numerical representations of the Earth's climate system. They simulate \
-    the interactions between the atmosphere, oceans, land surface, and ice. These models are based \
-    on fundamental physical, chemical, and biological processes and are used to understand past, \
-    present, and future climate conditions. Each source or model is typically associated with a \
-    specific research institution, center, or group. For instance, models like 'EC-Earth' are \
-    developed by a consortium of European institutes, while 'GFDL-CM4' is developed by the \
-    Geophysical Fluid Dynamics Laboratory (GFDL) in the United States.
+    Source of the dataset
+
+    Examples: "CanESM6-MR", "CR-CMIP-1-0-0"
+
+    The more precise meaning of source depends on the kind of dataset this is.
+    For model output, 'source' refers to a numerical representations of the Earth's climate system.
+    This source is the model which was used to generate the dataset.
+    Such models simulate the interactions between the atmosphere, oceans, land surface, and ice.
+    They are based on fundamental physical, chemical, and biological processes
+    and are used to understand past, present, and future climate conditions.
+    Each source or model is typically associated with a specific research institution, center, or group.
+    For instance, models like 'EC-Earth' are developed by a consortium of European institutes,
+    while 'GFDL-CM4' is developed by the Geophysical Fluid Dynamics Laboratory (GFDL) in the United States.
+
+    For model inputs i.e. forcings, the 'source' is a unique identifier
+    for the group that produced the data and its version.
+    This is a different convention from almost all other cases
+    (which really muddies the meaning of the term).
+    """
+
```

Comment:
stripped down to minimal set as advised by Matt see https://github.com/ESGF/esgf-vocab/pull/51#issuecomment-3432513075